File: packages/ts/src/catalog/signEvents.ts
/**
 * Generated catalog module — do not hand-edit; rerun the generator.
 *
 * Produced by `packages/ts/scripts/gen-catalog.mjs`.
 * Regenerate with `npm run gen:catalog` (runs automatically before `build`).
 *
 * Source data:
 *   - isa-platform/shared/go/events/registry.go
 */

/**
 * RapidSign webhook event types. The wire string is the EventBridge
 * `detail-type` value the platform emits.
 */
export enum SignEvent {

}

export const SignEventLabels: Readonly<Record<SignEvent, string>> = Object.freeze({

}) as Readonly<Record<SignEvent, string>>;

File: packages/csharp/src/Catalog/SignEvents.cs
// CATALOG-GEN: do not hand-edit; rerun packages/csharp/scripts/gen-catalog.mjs.
//
// Source data:
//   - isa-platform/shared/go/events/registry.go

using System.Collections.Generic;
using System.Collections.ObjectModel;

namespace Sah.Sdk.Catalog;

/// <summary>RapidSign webhook event types.</summary>
public enum SignEvent
{
    [WireValue("document.signed")] DocumentSigned,
}

/// <summary>Label per sign event.</summary>
public static class SignEventLabels
{
    private static readonly IReadOnlyDictionary<SignEvent, string> MAP = new ReadOnlyDictionary<SignEvent, string>(new Dictionary<SignEvent, string>
    {
        [SignEvent.DocumentSigned] = "DocumentSigned",
    });

    /// <summary>Get the label for an event.</summary>
    public static string Get(SignEvent e) => MAP.TryGetValue(e, out var v) ? v : string.Empty;
}
